neuro_san/internals/tool_factories/service_tool_factory_provider.py

```diff
@@ -46,6 +48,19 @@ def get_instance(cls):
             ServiceToolFactoryProvider.instance = ServiceToolFactoryProvider()
         return ServiceToolFactoryProvider.instance
```

Comment:
Another case of a questionable singleton. 
The right ownership model could obviate the need for the singleton.
(This one is not for this PR)